src/api/app/models/bs_request.rb

```diff
@@ -1199,14 +1199,16 @@ def _assignreview_update_reviews(reviewer, opts, new_review = nil)
     review_comment
   end
 
-  # TODO: Once the feature flag notifications_redesign is removed, remove this method if we decide to delete the "Tasks" page.
+  # drop all user caches, mainly for involed requests operations
   def update_cache
     target_package_ids = bs_request_actions.with_target_package.pluck(:target_package_id)
     target_project_ids = bs_request_actions.with_target_project.pluck(:target_project_id)
 
-    user_ids = Relationship.where(package_id: target_package_ids).or(
+    group_users = Relationship.where(package_id: target_package_ids).or(
       Relationship.where(project_id: target_project_ids)
-    ).groups.joins(:groups_users).pluck('groups_users.user_id')
+    ).groups.joins(:groups_users)
+
+    user_ids = group_users.pluck('groups_users.user_id')
```

Comment:
Hm, hey @adrianschroeter I don't see what you are trying to achieve, but just separating the commands, won't make them to run separated.  `where`, `select`, `order`, `group`, and other query methods return an instance of `ActiveRecord::Relation`, which represents the results of a SQL query. It doesn’t run the query until you try and access one of the records returned by it. So the query will just just be evaluated when you call `.pluck`